From d97e5e4c1e85c5a9ef7364a5c71a7a57e5ef14d1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marko=20Luk=C5=A1a?= <marko.luksa@gmail.com>
Date: Tue, 15 Feb 2022 17:03:46 +0100
Subject: Add mec to pilot image

When you build the images, the pilot image didn't include the mec binary. This caused the wash-cacher pod to not start.
---
 maistra.mk                    | 2 +-
 pilot/docker/Dockerfile.pilot | 1 +
 tools/istio-docker.mk         | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/maistra.mk b/maistra.mk
index f90194cf95..ec9b03c331 100644
--- a/maistra.mk
+++ b/maistra.mk
@@ -6,4 +6,4 @@ vendor:
 
 gen: vendor
 
-STANDARD_BINARIES += ./mec/cmd/mec
+AGENT_BINARIES += ./mec/cmd/mec
diff --git a/pilot/docker/Dockerfile.pilot b/pilot/docker/Dockerfile.pilot
index fe2c87eef9..ef4fc27390 100644
--- a/pilot/docker/Dockerfile.pilot
+++ b/pilot/docker/Dockerfile.pilot
@@ -16,6 +16,7 @@ FROM ${BASE_DISTRIBUTION:-debug}
 
 ARG TARGETARCH
 COPY ${TARGETARCH:-amd64}/pilot-discovery /usr/local/bin/pilot-discovery
+COPY ${TARGETARCH:-amd64}/mec /usr/local/bin/mec
 
 # Copy templates for bootstrap generation.
 COPY envoy_bootstrap.json /var/lib/istio/envoy/envoy_bootstrap_tmpl.json
diff --git a/tools/istio-docker.mk b/tools/istio-docker.mk
index c83482bb93..89705ea1a8 100644
--- a/tools/istio-docker.mk
+++ b/tools/istio-docker.mk
@@ -102,6 +102,7 @@ docker.pilot: BUILD_ARGS=--build-arg BASE_VERSION=${BASE_VERSION}
 docker.pilot: ${ISTIO_ENVOY_BOOTSTRAP_CONFIG_DIR}/envoy_bootstrap.json
 docker.pilot: ${ISTIO_ENVOY_BOOTSTRAP_CONFIG_DIR}/gcp_envoy_bootstrap.json
 docker.pilot: $(ISTIO_OUT_LINUX)/pilot-discovery
+docker.pilot: $(ISTIO_OUT_LINUX)/mec
 docker.pilot: pilot/docker/Dockerfile.pilot
 	$(DOCKER_RULE)
 
-- 
2.31.1

